src/Templates/TypeEditorTemplate/CustomTypeEditor/CustomControlLibrary.Design.Server/CodeDomSerializer/CustomPropertyStoreCodeDomSerializer.cs

```diff
@@ -49,16 +49,16 @@ internal class CustomPropertyStoreCodeDomSerializer : CodeDomSerializer
                 // It just traverses the property's object graph and creates the code for generating 
                 // the respective value types, enums, arrays, etc.
 
-                // To that end, we're _not_ getting this type's serializer, because then we would ending up with this exact serializer.
-                // This is not what we want. Instead, we want to serialize the graph of property type, so we're getting object's 
-                // default serializer...
+                // To that end, we're _not_ getting this type's serializer, because then we would ending up with running 
+                // this exact serializer. This is not what we want, we would end up in a recusion. Instead, we want to serialize 
+                // the graph of property type, so we're getting object's default serializer...
```

Comment:
Could you help me understand what this bit means "_...serialize the graph of property type..._"?

Reply:
Sure. Let's say, you have a complex, composed type. In that case, there are 2 ways to get that thing serialized:

a) You need to take care of ALWAYS instantiating the type, when the property gets read. Because our stock serializer is not able to do the instantiation and assign that instance to the property. It can, however, figure out what intrinsic types the complex property is composed of and generate the serialization code for that. I don't like this, since that means, such a property can never be `null`.

b) We use this Serializers. It's basically amending that missing instantiation part, and then it jumps into our stock serializer, traverses the type-graph of that complex property and generates the serialization code to assign the properties, which that complex type is composed of.

I also chose the latter to also have a sample how a custom CodeDomSerializer is provided per se.